src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -72,7 +72,7 @@ class RsPathReferenceImpl(
     }
 
     override fun advancedResolve(): BoundElement<RsElement>? =
-        advancedMultiResolve().singleOrNull()?.inner
+        advancedMultiResolve().singleOrNull { it.inner.element !is RsFunction }?.inner
```

Comment:
Actually, it breaks name resolution because path can refer to function. As a result, some test fail